src/actions/guides/frontend/internationalization.cr

```diff
@@ -0,0 +1,341 @@
+    ```
+    shards install
+    ```
+
+    ## Step 2 - Add localization ymls
+
+    first make a locales folder:
+    ```
+    mkdir config/locales
+    ```
+
+    then add at least one localization: i.e. English
+    ```
+    # config/locales/en.yml
+    en:
+      auth:
+        sign_in: "Sign-In"
+        sign_up: "Sign-up"
+        update_pwd: "Update Password"
+        reset_pwd_request: "Reset Password"
+      default:
+        page_name: Welcome
+      me:
+        email: E-Mail
+        profile: Profile
+        next: "Next, you may want to"
+        auth_guides: "Check out the Authentication Guides"
+        modify_page: "Modify this page"
+        after_signin: "Change where you go after sign in"
+    ```
+
+    and for example additional ones as needed: i.e. German
+    ```
+    # config/locales/de.yml
+    de:
+      auth:
+        sign_in: Anmeldung
+        sign_up: Anmelden
+        sign_out: Abmelden
+        update_pwd: "Kennwort aktualisieren"
+        reset_pwd_request: "Kennwort zurücksetzen"
+      default:
+        page_title: Wilkommen
+      me:
+        email: Mail
+        profile: Profil
+        next: "Zunächst bitte kontrolliere"
+        auth_guides: "Bitte kontrolliere die 'Authentication Guides'"
+        modify_page: "Um diese Seite zu Ändern"
+        after_signin: "Eine andere Seite nach anmelden"
+    ```
+
+    **NOTE:** All lang yml files need all the same keys defined. I18n shard is pretty finicky and its error messages aren't very helpful _(expect a little frustation getting the ymls correct and debugged)._
+
+    ## Step 3 - Configure i18n within Lucky
+    ```
+    # config/i18n.cr
+    I18n.backend = I18n::Backends::YAML.new.tap do |backend|
+      backend.load_paths << Dir.current + "/config/locales"
+      backend.load
+    end
+    ```
+
+    ## Step 4 - Add 'lang' to users table
+
+    generate a migration using:
+    ```
+    lucky db.migration AddLanguageToUser
+    ```
+
+    edit the file to look like (of course, the number will vary):
+    ```
+    # db/migrations/20191228100116_add_language_to_user.cr
+    class AddLanguageToUser::V20191228100116 < Avram::Migrator::Migration::V1
+      def migrate
+        alter table_for(User) do
+          add lang : String, default: "en"  # the appropriate default lang key
+        end
+      end
+
+      def rollback
+        alter table_for(User) do
+          remove :lang
+        end
+      end
+    end
+    ```
+
+    and of course migrate
+    ```
+    lucky db.migrate
+    ```
+
+    ## Step 5 - Add lang column to User model
+    ```
+    # src/models/user.cr
+    class User < BaseModel
+      # ...
+      table do
+        column lang : String
+        # ...
+      end
+      # ...
+    end
+    ```
+
+    ## Step 6 - Create a Translator class
+
+    TODO: finalize the best approach & ideally be ready to capture any Frontend JS overrides
+    TODO: autodect and autocreate the language list based on the config/locales yml files.
```

Comment:
```suggestion
    TODO: autodetect and autocreate the language list based on the config/locales yml files.
```

Reply:
I removed after I realized that the laguages need to be named in the selector & I don't see how that can be automated.